src/main/webapp/index.html

```diff
@@ -5,19 +5,36 @@
     <title>Meal Assistant</title>
     <link rel="stylesheet" href="search-meal-style.css"></link>
     <link rel="stylesheet" href="style.css"></link>
+    <script src="https://unpkg.com/vue"></script>
     <script src="search-meal-script.js"></script>
-  </head>
-  <body onload="searchMeal()">
-    <div id="search-block">
-      <form>
-        <label for="text">Meal Assistant</label>
-        <input type="text" id="text" name="text" placeholder="Search.." />
-        <input type="submit" />
-      </form>
-      <ul id="dishes-container"></ul>
+    <script src="script.js"></script>
+    <script type="module" src="main.js"></script>
+  </head>  
+  <body>
+    <nav class="topnav">
+        <b> <a href="/index.html"> Meal Assistant</a> </b>
+    </nav>
+    <div id="background">
+        <div id="search-block">
+            <div id="search-title">
+                Meal Assistant
+            </div>
+            <br>
+            <div id="search-bar-block"> 
+                <div id="search-bar"> 
+                    <div id="bar">
+                        <input v-on:keyup.enter="submit" v-model="request" type="text" id="text" name="text" placeholder="Search..." />
+                        <div v-on:click="submit" id="search-icon"></div>
+                    </div>
```

Comment:
This div is used twice. We could extract it to a Vue component:
https://vuejs.org/v2/guide/components.html

Reusable components are the most common way of structuring Vue's (or any framework's) code.